client/src/main/java/io/pravega/keycloak/client/KeycloakAuthzClient.java

```diff
@@ -40,9 +43,13 @@
 
     public static final String DEFAULT_PRAVEGA_CONTROLLER_CLIENT_ID = "pravega-controller";
     private static final int DEFAULT_TOKEN_MIN_TIME_TO_LIVE_SECS = 60;
+    private static final int DEFAULT_HTTP_MAX_RETRIES = 20;
+    private static final int DEFAULT_HTTP_INITIAL_DELAY_MS = 100;
 
     private final AuthzClient client;
     private final TokenCache tokenCache;
+    private int httpMaxRetries = DEFAULT_HTTP_MAX_RETRIES;
```

Comment:
I'd suggest making `httpMaxRetries` and `httpInitialDelayMs` final too, and setting these to default values in the constructor that does not take these as arguments. Doing that'd make this class immutable and threadsafe.

Reply:
agreed. fixed.